Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build Stage
+FROM golang:1.12 as build
+
+RUN apt-get update && apt-get install -y
+
+ARG VERSION="1.14-develop"
+
+WORKDIR /go/src/cloudsql-proxy
+COPY . .
+
+RUN go get ./cmd/cloud_sql_proxy && go build -ldflags "-X 'main.versionString=$VERSION'" \
```

Comment:
I think it would be safer if you don't rely on the `libc` from the distroless image, and use Go’s static compilation flags.

Most of the time people use this to truly statically compile Go binaries:

     go build -a -tags netgo -ldflags '-w -extldflags "-static"'  ...

Reply:
Why's it safer? The glibc dependency is the most tested and most used in production. Unless there's a concrete reason not to use it, I'd go with the defaults.

Reply:
The only reason would be if someone tries to use this binary like:

FROM this_image AS dep
FROM their_alpine
COPY -from=dep /this_binary /bin

And libc would cause issues

Reply:
I'm going to revert for now - it's probably safer for users to dl the binary from one of the links over pulling it out of this image.